ships/0006-surface-results-in-buildruns.md

```diff
@@ -0,0 +1,293 @@
+
+- The _.status_ subresource should provide a relationship between a source and the results emitted in it´s step definition.
+- The _.status_ subresource should be able to surface results emitted from different sources.
+
+Therefore the **proposal** is to introduce `_.status.sources_` subresource which holds a list of `SourcesResult`:
+
+```go
+type KeyValue map[string]string
+
+type Sources struct {
+  Results []SourceResult `json:"sources"`
+}
+
+type SourcesResult struct {
+  Name   string   `json:"name"`
+  Git    KeyValue `json:"git,omitempty"`
+  Http   KeyValue `json:"http,omitempty"` // A new field can be always introduced to support new types, e.g. bundle
+}
+```
+
+A BuildRun will surface these results as follows:
+
+```yaml
+status:
+  sources:
+  - name: default
+    git:
+      commit-sha: abcd
+```
+
+#### Build Results Output API Proposal
+
+Today, we only have a single output step in our strategies. There is no plan to modify this, therefore we propose here to introduce the _.status.output_ subresource in a BuildRun. This will require the following struct:
+
+```go
+type Output struct {
+  Results KeyValue `json:"output"`
+}
+```
+
+where output holds a map of key/values.
+
+```yaml
+ status:
+   output:
+     digest: sha:30213102
+```
+
+#### Proposal References
+
+The following go playground [example](https://play.golang.org/p/KLSwUJeiX0T) highlights the usage of both Sources and Output Build Resource Objects.
+
+### User Stories [optional]
+
+#### As a Strategy Author, I want to surface strategy higher-level information
+
+Via the `Strategy Results` we provide flexibility for Strategy Authors to decide on particular results they would
+like to surface. All they need to do is:
+
+- Define the result in their strategy, under `spec.results`, this consist of a name and a description.
+- Populate or Emit the result to `'$(results.<result-name>.path)'`
+
+#### As a Shipwright Build Developer, I want to surface Build meaningful results
+
+Shipwright Developers should conclude on which results defined at runtime will belong under `.status.sources` and `.status.output`. Today, we emit the following:
+
+- shp-image-size
+- shp-image-digest
+- shp-source-default-commit-sha
+
+therefore we should populate `.status` subresource with the following:
+
+```yaml
+ status:
+   sources:
+   - name: default
+     git:
+       commit-sha: cf156cfb0da9be144a40c7d5a110487e5c
+   output:
+     image-digest: sha:30213102
+```
+
+### As a Shipwright CLI developer, I want to consume surfaced results
+
+Workflows on top of Shipwright should be able to consume higher-level data and present that in different ways. As long as they consume from `.status.sources`, `.status.output` or `.status.results`.
+
+### Test Plan
+
+It should include:
+
+- Integration-tests
+- Unit-tests
+
+### Release Criteria
+
+Should be available before release v1.0.0
+
+### Risks and Mitigations
+
+- `Strategy Results` could pile up, ending-up in heavy BuildRun objects.
```

Comment:
We will need to document the length limitation behind this which is that Tekton uses container termination messages to transport those values out of a container into a custom resource. And termination messages have a length limitation.

The 4k mentioned here: https://github.com/tektoncd/pipeline/blob/main/docs/tasks.md#emitting-results